client.go

```diff
@@ -207,40 +216,36 @@ func (c Client) GetDatasetState(datasetID string) (*DatasetState, error) {
 }
 
 /*
-CreateDataset creates a volume in Flocker, waits for it to be ready and
+CreateDS creates a volume in Flocker, waits for it to be ready and
 returns the dataset id.
 
 This process is a little bit complex but follows this flow:
 
 1. Find the Flocker Control Service UUID
-2. Try to create the dataset
-3. If it already exists an error is returned
-4. If it didn't previously exist, wait for it to be ready
+2. If it already exists an error is returned
+3. If it didn't previously exist, wait for it to be ready
 */
-func (c Client) CreateDataset(metaName string) (*DatasetState, error) {
+func (c *Client) CreateDataset(options *CreateDatasetOptions) (datasetState *DatasetState, err error) {
 	// 1) Find the primary Flocker UUID
-	// Note: it could be cached, but doing this query we health check it
-	primary, err := c.GetPrimaryUUID()
-	if err != nil {
-		return nil, err
+	// Note: it could be cached, but doing this query we health check iti
+	if options.Primary == "" {
+		options.Primary, err = c.GetPrimaryUUID()
```

Comment:
:question: I see that this is refactoring rather than a change but it looks like we're setting the primary to the current node....and the current node is identified by matching on  IP address...which has caused problems in the past...because when running `flocker-dataset-agent` in a container you get the container IP address unless you run it with `--net host`.

We should think about how / whether Flocker could report the Kubernetes node ID rather than generate its own.